controllers/nodehealthcheck_controller_test.go

```diff
@@ -315,6 +315,34 @@ var _ = Describe("Node Health Check CR", func() {
 			})
 		})
 
+		Context("control plane nodes", func() {
+			When("two control plane nodes are unhealthy, just one should be remediated", func() {
+				BeforeEach(func() {
+					objects = newNodes(2, 1, true)
+					objects = append(objects, newNodes(1, 5, false)...)
+					underTest = newNodeHealthCheck()
+					remediationTemplate := newRemediationTemplate()
+					objects = append(objects, underTest, remediationTemplate)
+				})
+
+				It("creates a one remediation CR for control plane node", func() {
+					Expect(reconcileError).NotTo(HaveOccurred())
+
+					cr := newRemediationCR("")
+					crList := &unstructured.UnstructuredList{Object: cr.Object}
+					Expect(reconciler.Client.List(context.Background(), crList)).To(Succeed())
+
+					Expect(len(crList.Items)).To(BeNumerically("==", 2), "expected 2 remediations, one for control plane, one for worker")
+					Expect(crList.Items).To(ContainElements(
+						// the unhealthy worker
+						HaveField("Object", HaveKeyWithValue("metadata", HaveKeyWithValue("name", "unhealthy-node-1"))),
```

Comment:
NIT: I think it would be nicer to explicitly use the word `worker` like we use for  `control-plane` at line 340, rather than leave it empty.

Reply:
ok